play-services-maps/core/mapbox/src/main/kotlin/org/microg/gms/maps/mapbox/GoogleMap.kt

```diff
@@ -848,9 +848,16 @@ class GoogleMapImpl(context: Context, var options: GoogleMapOptions) : AbstractG
     }
 
     fun getMapAsync(callback: IOnMapReadyCallback) {
+        var copyList: ArrayList<IOnMapReadyCallback>
         synchronized(mapLock) {
-            userOnInitializedCallbackList.add(callback)
+            copyList = ArrayList(userOnInitializedCallbackList)
         }
+
+        copyList.add(callback)
```

Comment:
Shouldn't this still all happen within one synchronized block? Otherwise, if `getMapAsync` is invoked twice at the same time, it may interleave in such a way that only one callback is stored.

Also, what would probably be much easier, would be to use the `java.util.concurrent.CopyOnWriteArrayList` - it handles this internally for you.

I do see the issue though, that if `getMapAsync` is invoked from within the map ready callback, that `getMapAsync` will never get it's callback invoked (as the clearing of the callback list is not tied to the actual list of callbacks that was invoked - that's not a new issue though). The better approach thus could be to change the calling code to create a copy of the list before invoking the callbacks, then clear the original list and then start invoking the callbacks from the copy. As the list that is iterated is a local copy, no ConcurrentModificationException could happen either and we wouldn't see the issue of callbacks getting never called.